packages/babel-parser/test/fixtures/typescript/types/typeof-private/input.ts

```diff
@@ -0,0 +1,23 @@
+class C {
+    #a = 'a';
+
+    constructor() {
+        const a: typeof this.#a = ''; // Ok
+        const b: typeof this.#a = 1;  // Error
+    }
+}
+
+const c = new C();
+const a: typeof c.#a = '';
```

Comment:
TS will throw "Property '#a' is not accessible outside class 'C' because it has a private identifier." in this case. We should call `classScope.usePrivateName` after private name is parsed as a TS entity name.